### PR TITLE
Fix search results don't automatically update on editor changes, if I move the Search view to another location

### LINKED ISSUE #215437: Search results don't automatically update on editor changes, if I move the Search view to another location
1. Open any file
2. Search for `foo`, see some results
3. Type `foo` in the editor, see that result shows up in the search view
4. Delete `foo`, see that result disappears
5. Move the search view below the File Explorer
6. Repeat steps 3-4

:bug: Search results no longer automatically update

### LINKED ISSUE #225064: Search results are not retained after moving search view to a different location
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

1. Search `foo` for some results
2. Move search view to a different location
3. 🐛  Search results from step 1 are not displayed

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #215437
Fixes #225064

### CODE REVIEW

**Review by BaljeettheBro — APPROVED** [review]

**Review by BaljeettheBro — APPROVED** [review]

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 1]
```diff
@@ -299,6 +299,8 @@ export class SearchView extends ViewPane {
 				this.searchWidget.prependReplaceHistory(restoredHistory.replace);
 			}
 		}));
+
+		this.changedWhileHidden = this.hasSearchResults();
```
Comment (andreamah) [inline src/vs/workbench/contrib/search/browser/searchView.ts:303]: Is this needed for this PR?

**Review by andreamah — CHANGES_REQUESTED** [review]
Nice catch! just one comment

File: src/vs/workbench/contrib/search/browser/searchView.ts (hunk 1 above)
Comment (jeanp413) [reply src/vs/workbench/contrib/search/browser/searchView.ts:303]: yes it has to be initialized with the proper value

Comment (andreamah) [reply src/vs/workbench/contrib/search/browser/searchView.ts:303]: That's fine. I just don't really understand how the two changes that you made are linked. I tried testing without this change above and the bug was still resolved. It seemed that the bug was primarily fixed by not registering the `searchViewModelWorkbenchService`

Comment (jeanp413) [reply src/vs/workbench/contrib/search/browser/searchView.ts:303]: - With this line change, if you had search results already displayed, after you move the search view it still render them.
- Without this line change,  if you had search results already displayed, after you move the search view will be empty.

Comment (andreamah) [reply src/vs/workbench/contrib/search/browser/searchView.ts:303]: Oh okay, that's a different bug. I interpreted the original bug as the entries not updating once you move the search viewlet. However, this seems to be that the information in the search viewlet does not retain itself across views. Do you mind creating an issue for that and linking that to this PR?

Comment (jeanp413) [reply src/vs/workbench/contrib/search/browser/searchView.ts:303]: Done 👍

**Review by andreamah — APPROVED** [review]
Awesome, thanks for the contribution!

**Review by bpasero — APPROVED** [review]

**Pull request merged**